src/components/PDFView/index.js

```diff
@@ -105,10 +162,32 @@ class PDFView extends Component {
         this.props.onToggleKeyboard(isKeyboardOpen);
     }
 
+    /**
+     * Renders a specific page based on its index.
+     * It includes a wrapper to apply virtualized styles.
+     * @param {Number} index
+     * @param {Object} style
+     * @returns {JSX.Element}
+     */
+    renderPage({index, style}) {
+        const pageWidth = this.calculatePageWidth();
+
+        return (
+            <View style={style}>
+                <Page
+                    key={`page_${index}`}
+                    width={pageWidth}
+                    pageIndex={index}
+                    // This needs to be empty to avoid multiple loading texts which show per page and look ugly
+                    // See https://github.com/Expensify/App/issues/14358 for more details
+                    loading=""
+                />
+            </View>
+        );
+    }
+
     render() {
-        const pdfContainerWidth = this.state.windowWidth - 100;
-        const pageWidthOnLargeScreen = pdfContainerWidth <= variables.pdfPageMaxWidth ? pdfContainerWidth : variables.pdfPageMaxWidth;
-        const pageWidth = this.props.isSmallScreenWidth ? this.state.windowWidth : pageWidthOnLargeScreen;
+        const pageWidth = this.calculatePageWidth();
```

Comment:
I see that pageWidth is being calculated 2 times on render: one here, and another in `renderPage()`. Can we pass this value here to `renderPage()` to reuse it?

Reply:
Good catch! 
I can do via currying, something like that:
```javascript
    renderPage(pageWidth) {
        return ({index, style}) => (
            <View style={style}>
                <Page
                    key={`page_${index}`}
                    width={pageWidth}
                    pageIndex={index}
                    // This needs to be empty to avoid multiple loading texts which show per page and look ugly
                    // See https://github.com/Expensify/App/issues/14358 for more details
                    loading=""
                />
            </View>
        );
    }
    
    render() {
        const pageWidth = this.calculatePageWidth();
        ...
        <List
           ...
       >
          {this.renderPage(pageWidth)}
      </List>
``` 

Does it follow the project rules?

Reply:
@rezkiy37 LGTM!

Reply:
cool!